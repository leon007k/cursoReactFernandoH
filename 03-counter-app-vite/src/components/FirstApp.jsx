import PropTypes from "prop-types";
import { ListOfMessages } from "../const";

function getListOfMessages(id) {
  const messages = ListOfMessages.find(message => message.id === id);
  return { autorMessage: messages.name, messages: messages.message };
}

export default function FirstApp({ title = 'Hello World from FirstApp', subtitle = 'This is a subtitle' }) {

  const { autorMessage, messages } = getListOfMessages(2);

  return (
    <>
      <h2 data-testid="test-title">{title}</h2>
      <h3>{subtitle}</h3>
      <h3>{subtitle}</h3>
      <p>
        Autor: {autorMessage}
        <br />
        Mensaje: {messages}
      </p>

    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

/*
  * Se remueven por el warning:
  ! FirstApp: Support for defaultProps will be removed from function components in a future major release.
  ! Use JavaScript default parameters instead
FirstApp.defaultProps = {
  title: 'Hello World from FirstApp',
  subtitle: 'This is a subtitle'
} */
