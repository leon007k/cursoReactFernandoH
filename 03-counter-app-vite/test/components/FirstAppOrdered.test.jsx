import { render, screen } from '@testing-library/react';
import FirstApp from '../../src/components/FirstApp';

describe('Pruebas en <FirstApp />', () => {

  const title = 'Hola, soy Goku!!'
  const subtitle = 'Soy un subtitulo'

  test('debe hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  })

  test('debe mostrar el mensaje "Hola, soy Goku!!"', () => {
    render(<FirstApp title={title} />);

    // * screen -> Objeto a renderizar. Contiene todo lo renderizado
    expect(screen.getByText(title)).toBeTruthy();
    // * Verificamos que el elemento con el testId 'test-title' contenga el texto 'Hola, soy Goku!!'
    //expect(screen.getByTestId('test-title').innerHTML).toBe(title)
  })

  test('Debe de mostrar el titulo en un h2', () => {
    render(<FirstApp title={title} />);

    // * getByRole -> Busca un elemento por su rol
    // * heading -> Busca un elemento de tipo h1, h2, h3, h4, h5, h6
    // * level: 2 -> Busca un elemento h2
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe(title)
  })

  test('Debe de mostrar dos subtitulos', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);

    expect(screen.getAllByText(subtitle).length).toBe(2);
  })
});