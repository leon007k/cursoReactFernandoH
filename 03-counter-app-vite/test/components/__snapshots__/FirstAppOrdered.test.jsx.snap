// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h2
    data-testid="test-title"
  >
    Hola, soy Goku!!
  </h2>
  <h3>
    This is a subtitle
  </h3>
  <h3>
    This is a subtitle
  </h3>
  <p>
    Autor: 
    Jane Doe
    <br />
    Mensaje: 
    Hi there!
  </p>
</div>
`;
